src/widget/list_select.rs

```diff
@@ -0,0 +1,465 @@
+
+use {
+    color,
+    Color,
+    Colorable,
+    event,
+    FontSize,
+    IndexSlot,
+    Labelable,
+    NodeIndex,
+    Positionable,
+    Scalar,
+    List,
+    Button,
+    Borderable,
+};
+use widget::{self, Widget};
+use std::fmt::Display;
+
+use std;
+
+/// Displays a given Vec<T> where T: Display as a selectable List. Its reaction is triggered upon
+/// selection of a list item.
+pub struct ListSelect<'a, T, F> where F: FnMut(Event), T: Display+'a {
+    entries: &'a [T],
+    selected: &'a mut [bool],
+    common: widget::CommonBuilder,
+    maybe_react: Option<F>,
+    multiple_selections: bool,
+    style: Style,
+}
+
+widget_style!{
+    /// Styling for the ListSelect, necessary for constructing its renderable Element.
+    style Style {
+        /// Size of text font in list
+        - font_size: FontSize { theme.font_size_medium }
+        /// The background color of the unselected entries.
+        - color: Color { theme.shape_color }
+        /// The background color of the selected entries.
+        - selected_color: Color { theme.label_color }
+        /// Color of the item text, when not selected.
+        - text_color: Color { theme.shape_color }
+        /// Color of the item text, when selected.
+        - selected_text_color: Color { color::BLACK }
+        /// Width of the border surrounding the widget
+        - border: Scalar { theme.border_width }
+        /// The color of the border.
+        - border_color: Color { theme.border_color }
+        /// Font size for the item labels.
+        - label_font_size: FontSize { theme.font_size_medium }
+        /// Auto hide the scrollbar or not
+        - scrollbar_auto_hide: bool { true }
+    }
+}
+
+/// Represents the state of the ListSelect.
+#[derive(PartialEq, Clone, Debug)]
+pub struct State {
+    list_idx: IndexSlot,
+    /// Tracking index of last selected entry that has been pressed in order to
+    /// perform multi selection when `SHIFT` or `ALT`(Mac) / 'CTRL'(Other OS) is held.
+    last_selected_entry: Option<usize>,
+}
+
+pub use self::Event::{SelectEntry, SelectEntries, DoubleClick, KeyPress};
+
+/// The kind of events that the `ListSelect` may `react` to.
+/// Provides tuple(s) of index in list and string representation of selection
+#[derive(Clone, Debug)]
+pub enum Event {
+    SelectEntry((usize, String)),
+    SelectEntries(Vec<(usize, String)>),
+    DoubleClick((usize, String)),
```

Comment:
It might be a little more idiomatic to change this variant to
```rust
DoubleClick(usize, String),
```
rather than using a tuple (same goes for the `SelectEntry` variant)?